2.7.0/final.md

```diff
@@ -0,0 +1,958 @@
+  Ring Attention ([#131351](https://github.com/pytorch/pytorch/pull/131351)) and an AllGather-based approach ([#132820](https://github.com/pytorch/pytorch/pull/132820)) where the all-gather is issued before the first local SDPA
+  and the subsequent local SDPAs will have to wait until the all-gather completes, and offered a user API ([#142093](https://github.com/pytorch/pytorch/pull/142093)) to select the desired approach. The implementation
+  currently supports three SDPA kernels: `SDPBackend.FLASH_ATTENTION`, `SDPBackend.EFFICIENT_ATTENTION`, and `SDPBackend.CUDNN_ATTENTION` ([#148537](https://github.com/pytorch/pytorch/pull/148537)). We also
+  verified that our Context Parallel implementation is compatible with other parallelisms and `torch.compile`.
+### c10d
+- Implemented ncclCommInitRankScalable (merging #136789) ([#144794](https://github.com/pytorch/pytorch/pull/144794))
+### Distributed Checkpoint (DCP)
+- Cache save plans: to mitigate overhead from planning steps ([#147116](https://github.com/pytorch/pytorch/pull/147116), [#147343](https://github.com/pytorch/pytorch/pull/147343))
+- Build a storage reader/writer to write checkpoints in HF format ([#148089](https://github.com/pytorch/pytorch/pull/148089))
+
+## CUDA
+- RTX50 Blackwell Support codegen ([#145270](https://github.com/pytorch/pytorch/pull/145270))
+- Make `torch.cuda.gds` APIs public ([#147120](https://github.com/pytorch/pytorch/pull/147120))
+
+## MPS
+
+## ROCm
+- CK Memory-Efficient Attention (attention bias support) (#147778)
+- CK Flash Attention Backend (#143695)
+- Enhanced Windows support for PyTorch on ROCm (#148563, #144098)
+- Support for gfx1102 arch (Navi33) in wheel builds (#147761)
+- hipblaslt rowwise f8 gemm (#144432)
+
+## XPU
+- Add AOT Inductor support for Intel GPU ([#140269](https://github.com/pytorch/pytorch/pull/140269), [#140664](https://github.com/pytorch/pytorch/pull/140664), [#149175](https://github.com/pytorch/pytorch/pull/149175))
+- Support `torch.compile` on Windows Platform for XPU ([#147637](https://github.com/pytorch/pytorch/pull/147637), [#144316](https://github.com/pytorch/pytorch/pull/144316))
+- Support SYCL with `torch.utils.cpp_extension` APIs ([#132945](https://github.com/pytorch/pytorch/pull/132945))
+- Enhance Intel GPU performance on PyTorch 2 Export Post Training Quantization ([#136753](https://github.com/pytorch/pytorch/pull/136753), [#135465](https://github.com/pytorch/pytorch/pull/135465),[#135337](https://github.com/pytorch/pytorch/pull/135337), [#135189](https://github.com/pytorch/pytorch/pull/135189))
+- Enable windows Kineto profiler([#148319](https://github.com/pytorch/pytorch/pull/148319))
+- Enable TF32 support for XPU based on oneDNN backend ([#137570](https://github.com/pytorch/pytorch/pull/137570))
+
+## torch.dynamo
```

Comment:
neither torch.dynamo nor torch.inductor exist as submodule. Can we use a less confusing notation here?

Reply:
Anything that doesn't look like a python submodule is fine for me !